atlp-rwanda/ecommerce-app-predators-bn
# #3 Create Checkout Flow

#### What does this PR do?
- Creates and order after checkout
- Calculates discount from applied coupons
#### Description of Task to be completed

> The backend needs to handle the checkout process, including validating and processing the order, updating inventory, and updating the customer's account.

#### How should this be manually tested?
- Checkout in branch `ft-checktout-3` and pull most recent changes
- Start the server by running `npm run start`
- Make sure you have a user account already, and few products in your cart before proceeding checkout.
- Make a POST request to this route `/api/checkout` with a requestBody containing the following properties `country, street_address, town, state, post_code, email` AKA billing information. Wait for a successful response. Kudos !!! You have just initiated an order.

#### Relevant pivotal tracker stories?
- [Card #3](https://trello.com/c/LUfLttI6)
